stacker_blueprints/dynamodb.py

```diff
@@ -0,0 +1,162 @@
+from stacker.blueprints.base import Blueprint
+
+from troposphere import (
+    dynamodb2,
+    Ref,
+    GetAtt,
+    Output,
+)
+
+from troposphere.route53 import RecordSetType
+from . import util
+
+MAX_GSI_VALUE = 5
+
+
+def prep_schemata(config):
+    try:
+        schemata = []
+        for schema in config["KeySchema"]:
+            schemata.append(dynamodb2.KeySchema(**schema))
+
+    except KeyError:
+        raise KeyError(
+            "The key schema is required for the creation of a DynamoDB table."
+        )
+
+    return schemata
+
+
+def prep_projection(config):
+    return dynamodb2.Projection(**config["Projection"])
+
+
+def prep_throughput(config):
+    try:
+        return dynamodb2.ProvisionedThroughput(
+            **config["ProvisionedThroughput"]
+        )
+    except KeyError:
+        raise KeyError(
+            "ProvisionedThroughput values are required for the creation of " +
+            "a DynamoDB table or index."
+        )
+
+
+def prep_config(raw_config):
+    prepped_config = {}
+
+    dynamodb_table_properties = [
+        "AttributeDefinitions",
+        "GlobalSecondaryIndexes",
+        "KeySchema",
+        "LocalSecondaryIndexes",
+        "ProvisionedThroughput",
+        "StreamSpecification",
+    ]
+
+    util.check_properties(
+        raw_config,
+        dynamodb_table_properties,
+        "DynamoDB",
+    )
+
+    # AttributeDefinitions are required, so raise an IndexError if this
+    # doesn't work.
+    try:
+        attributes = []
+        for attribute in raw_config["AttributeDefinitions"]:
+            attributes.append(dynamodb2.AttributeDefinition(**attribute))
+
+        prepped_config["AttributeDefinitions"] = attributes
+    except KeyError:
+        raise KeyError(
+            "Attribute definitions are required for the creation of a " +
+            "DynamoDB table."
+        )
+
+    if "GlobalSecondaryIndexes" in raw_config:
+        # AWS limits us to 5 GSIs.  Check for that and bail if there's more.
+        if len(raw_config["GlobalSecondaryIndexes"]) > MAX_GSI_VALUE:
+            raise ValueError(
+                "A DynamoDB table can only have a maximum of 5 GSIs."
+            )
+
+        gsis = []
+        for gsi in raw_config["GlobalSecondaryIndexes"]:
+            gsi["KeySchema"] = prep_schemata(gsi)
+            gsi["Projection"] = prep_projection(gsi)
+            gsi["ProvisionedThroughput"] = prep_throughput(gsi)
+            gsis.append(dynamodb2.GlobalSecondaryIndex(**gsi))
+
+        prepped_config["GlobalSecondaryIndexes"] = gsis
+
+    prepped_config["KeySchema"] = prep_schemata(raw_config)
+
+    if "LocalSecondaryIndexes" in raw_config:
+        # Another limit of 5.  Check and bail if more than that.
+        if len(raw_config["LocalSecondaryIndexes"]) > MAX_GSI_VALUE:
```

Comment:
Did you intend to use the same constant for both LSI and GSI?  Probably worth making them different, only to avoid confusion, even if the underlying value is the same at the moment.